jira/client.go

```diff
@@ -0,0 +1,216 @@
+package jira
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/urlutil"
+)
+
+const (
+	apiEndPoint     = "/rest/api/2/issue/"
+	commentEndPoint = "/comment"
+)
+
+// Client ...
+type Client struct {
+	token   string
+	client  *http.Client
+	headers map[string]string
+	url     string
+}
+
+// Comment ...
+type Comment struct {
+	Content string
+	IssuKey string
+}
+
+type response struct {
+	issueKey string
+	err      error
+}
+
+func (resp response) String() string {
+	respValue := map[bool]string{true: colorstring.Green("SUCCES"), false: colorstring.Red("FAILED")}[resp.err == nil]
+	return fmt.Sprintf("Posting comment to - %s - : %s", resp.issueKey, respValue)
+}
+
+// -------------------------------------
+// -- Public methods
+
+// NewClient ...
+func NewClient(token, requestURL string) *Client {
```

Comment:
baseURL instead of request url. it is confusing, do we create new Client for every request?